lxd/storage_zfs.go

```diff
@@ -625,22 +625,31 @@ func (s *storageZfs) ImageCreate(fingerprint string) error {
 		return err
 	}
 
+	cleanup := func(err error) error {
+		if zerr := s.zfsDestroy(fs); zerr != nil {
+			err = fmt.Errorf("%s  During cleanup: %s", err, zerr)
+		}
+		if shared.PathExists(subvol) {
+			if oserr := os.Remove(subvol); oserr != nil {
```

Comment:
> But if the directory is empty, then I don't see how this fixes the
> problem in #2194; isn't that about leaving stuff around that actually
> takes up space?
That's my understanding as well, hence my theory that zfsDestroy is failing.  If zfsDestroy fails I don't think there's much that can be done other than propagating the cleanup error to the user.